File: backend/models/Review.js
const mongoose = require('mongoose');

const reviewSchema = new mongoose.Schema({
  car: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Car',
    required: true
  },
  customer: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  booking: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Booking',
    required: true
  },
  rating: {
    type: Number,
    required: [true, 'Please provide a rating'],
    min: 1,
    max: 5
  },
  comment: {
    type: String,
    required: [true, 'Please provide a comment'],
    maxlength: 500
  },
  cleanliness: {
    type: Number,
    min: 1,
    max: 5
  },
  comfort: {
    type: Number,
    min: 1,
    max: 5
  },
  performance: {
    type: Number,
    min: 1,
    max: 5
  },
  images: [{
    type: String
  }],
  helpful: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User'
  }],
  reported: {
    type: Boolean,
    default: false
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

// Calculate average rating for car after saving review
reviewSchema.post('save', async function() {
  const Car = mongoose.model('Car');
  const reviews = await mongoose.model('Review').find({ car: this.car });
  const avgRating = reviews.reduce((sum, review) => sum + review.rating, 0) / reviews.length;
  await Car.findByIdAndUpdate(this.car, { rating: avgRating });
});

module.exports = mongoose.model('Review', reviewSchema);

File: backend-java/src/main/java/com/carrental/model/Review.java
package com.carrental.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@Document(collection = "reviews")
public class Review {

    @Id
    private String id;
    
    private String carId; // Car ObjectId
    
    private String customerId; // User ObjectId
    
    private String bookingId; // Booking ObjectId
    
    private Integer rating; // 1 to 5
    
    private String comment;
    
    private Integer cleanliness; // 1 to 5
    
    private Integer comfort; // 1 to 5
    
    private Integer performance; // 1 to 5
    
    private List<String> images;
    
    private List<String> helpfulUserIds; // Array of User ObjectIds indicating helpfulness
    
    private boolean reported = false;
    
    private Date createdAt = new Date();
}
